src/IonParserTextRaw.ts

```diff
@@ -687,6 +687,7 @@ export class ParserTextRaw {
       ch = this._read_after_whitespace(true);
       if (ch != CH_CL) {
         this._error("expected ':'");
+        return;
```

Comment:
what about throw new Error ?

Reply:
It goes [here](https://github.com/amzn/ion-js/blob/master/src/IonParserTextRaw.ts#L987) to deal with errors. That's why returning with the error set.

Reply:
okay got it, thanks